backend/editor/entries.py

```diff
@@ -143,14 +180,44 @@ def create_project(self, description):
             SET n.taxonomy_name = $taxonomy_name
             SET n.branch_name = $branch_name
             SET n.description = $description
+            SET n.status = $status
+            SET n.created_at = datetime() 
         """
         params = {
             'project_name' : self.project_name,
             'taxonomy_name' : self.taxonomy_name,
             'branch_name' : self.branch_name,
-            'description' : description
+            'description' : description,
+            'status' : "OPEN"
         }
         result = get_current_session().run(query, params)
+    
+    def close_project(self):
+        """
+        Helper function to close a Taxonomy Editor project and updates project status as "CLOSED"
+        """
+        query = """
+            MATCH (n:PROJECT)
+            WHERE n.id = $project_name
+            SET n.status = $status
+        """
+        params = {
+            'project_name' : self.project_name,
+            'status' : "CLOSED"
+        }
+        result = get_current_session().run(query, params)

```

Comment:
Picky again: I feel like this is a bit risky to call current session here while all over the place we use current transaction.

One way to do it better might be to have a NewTransactionCtx in graph_db.py defined like:

```python
@contextlib.contextmanager
def NewTransactionCtx():
    """
    Transaction context will set global transaction "txn" for the code in context
    Use it to open a new transaction in current session
    """
    global txn
    with get_current_session().begin_transaction() as _txn:
        txn.set(_txn)
        yield txn
    txn.set(None)
```
and in code upper to close_project, (I prefer there, as it is the code, closing the primary transaction in the first place), use it:
```python3
        with NewTransactionCtx():
            self.close_project()
```

BTW the TransactionCtx could in turn call this, instead of duplicating code :-)

Reply:
Yep, I used the TransactionCtx again and changed everything to use transactions. :)